NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -1,140 +1,287 @@
+import cr_pulse_interpolator.signal_interpolation_fourier
 import logging
-logger = logging.getLogger('coreas')
+import copy
+import h5py
+logger = logging.getLogger('NuRadioReco.coreas')
+logger.setLevel(logging.INFO)
 
 warning_printed_coreas_py = False
 
 conversion_fieldstrength_cgs_to_SI = 2.99792458e10 * units.micro * units.volt / units.meter
 
-
-def get_angles(corsika):
+def get_angles(corsika, declination):
     """
-    Converting angles in corsika coordinates to local coordinates
+    Converting angles in corsika coordinates to local coordinates. 
+    
+    Corsika positiv x-axis points to the magnetic north, NuRadio coordinates positiv x-axis points to the geographic east.
+    Corsika positiv y-axis points to the west, NuRadio coordinates positiv y-axis points to the geographic north.
+    Corsika z-axis points upwards, NuRadio z-axis points upwards.
+
+    Corsikas zenith angle of a particle trajectory is defined between the particle momentum vector and the negativ z-axis, meaning that
+    the particle is described in the direction where it is going to. The azimuthal angle is described between the positive x-axis and 
+    the horizontal component of the particle momentum vector (i.e. with respect to the magnetic north) proceeding counterclockwise. 
+    
+    NuRadio describes the particle zenith and azimuthal angle in the direction where the particle is coming from. 
+    Therefore the zenith angle is the same, but the azimuthal angle has to be shifted by 180 + 90 degrees. The north has to be shifted by 90 degrees plus difference in geomagetic and magnetic north.
+
+
+    Parameters
+    ----------
+    corsika : hdf5 file object
+        the open hdf5 file object of the corsika hdf5 file
+    declination : float
+        declination of the magnetic field
     """
     zenith = np.deg2rad(corsika['inputs'].attrs["THETAP"][0])
-    azimuth = hp.get_normalized_angle(3 * np.pi / 2. + np.deg2rad(corsika['inputs'].attrs["PHIP"][0]))
+    azimuth = hp.get_normalized_angle(3 * np.pi / 2. + np.deg2rad(corsika['inputs'].attrs["PHIP"][0]) + declination)  # TODO: check if sign of declination is correct is correct
 
     Bx, Bz = corsika['inputs'].attrs["MAGNET"]
     B_inclination = np.arctan2(Bz, Bx)
 
     B_strength = (Bx ** 2 + Bz ** 2) ** 0.5 * units.micro * units.tesla
 
     # in local coordinates north is + 90 deg
-    magnetic_field_vector = B_strength * hp.spherical_to_cartesian(np.pi * 0.5 + B_inclination, 0 + np.pi * 0.5)
+    magnetic_field_vector = B_strength * hp.spherical_to_cartesian(np.pi * 0.5 + B_inclination, declination + np.pi * 0.5)  # TODO: check if sign of declination is correct is correct
 
     return zenith, azimuth, magnetic_field_vector
 
+def get_geomagnetic_angle(zenith, azimuth, magnetic_field_vector):
+    """
+    Calculates the angle between the geomagnetic field and the shower axis.
+    Parameters
+    ----------
+    zenith : float
+        zenith angle in radians
```

Comment:
radiants is the default units. You mean I should add a /units.rad?

Reply:
Yes, in case the internal system would change at some point :)

Reply:
I realised that I am not being very clear here, sorry. What I was specifically pointing to is that fact that you assume the zenith angle is given in radians (according to the docstring). But I would not do that/write it like that, ie what we really expect is the angle given in internal units. And then later, when calling the radiotools functions, we should indeed add `/ units.rad` to explicitly convert to radians (in case the internal units would ever change)